### ISSUE TITLE
Expel from a group

### ISSUE THREAD

**Issue by evanp** [swicg/groups#5]
"As a group administrator, I want to expel a member from a group, so that they can no longer participate as a member."

**Comment by mayel** [comment]
Is expelling the same as blocking? see https://github.com/swicg/groups/issues/29

**Comment by bumblefudge** [comment]
no, blocking is a per-user delivery/display affordance (show me no posts involving this user), while expelling means remove them from the group canonically/for everyone

**Comment by bumblefudge** [comment]
+1 to this being in the baseprofile, i wouldn't implement base if some implementations i interact with won't let me expel THEIR users for lack of extension support :D

**Comment by mayel** [comment]
> blocking is a per-user delivery/display affordance (show me no posts involving this user), while expelling means remove them from the group canonically/for everyone

I meant that the group could also block an actor or instance group-wide (so they cannot participate until unblocked), which I think is useful for moderation (eg. giving a member a "forced break" from the group while working through a report)

**Comment by trwnh** [comment]
@mayel I think that "permission to post" can be dynamically granted or removed temporarily. This concept is also known as a "time out". That would be a different issue, though -- #29 is closer to that. This issue #5 is about removing them from the group entirely. You can think of Join / Leave / Invite / Expel as a 2x2 matrix of actions:

x | Your action... | Someone else's action...
--- | --- | ---
...makes you a member | Join (#1) | Invite/Add (#4)
...makes you not a member | Leave (#2) | Expel/Remove (#5, this issue)

**Comment by TallTed** [comment]
`Expel` seems to rely on a previous `join`, with or without `invite`.

`Ban`, whether temporarily or permanently, might be a better label for this action.

**Comment by trwnh** [comment]
@TallTed "Ban" would be #29

**Comment by evanp** [comment]
To expel someone from a group ([Expel from a group][5]), an actor sends an `Remove` activity for the actor being expelled. The sending actor must have authorization to expel other actors.

```json
{
  "@context": "https://www.w3.org/ns/activitystreams",
  "id": "https://example.com/activities/1234567890",
  "actor": "https://example.com/groups/world-group/admins/3456789012",
  "type": "Remove",
  "object": {
    "id": "https://example.com/actors/27182818284",
    "type": "Person",
    "name": "Removed member"
  },
  "target": {
    "type": "OrderedCollection",
    "id":  "https://example.com/groups/world-group/members",
    "name": "World Group members"
  },
  "to": [
    {
      "type": "Group",
      "id": "https://example.com/groups/world-group",
      "name": "World Group"
    },
    {
      "id": "https://example.com/actors/27182818284",
      "type": "Person",
      "name": "Removed member"
    }
  ]
}
```

🚪 **Alternative** Other options exist for removing a member, such as using a `Reject` with their original `Join` activity, or using a `Block` activity.

❓ **Question:** Who has authorization to expel a member? The owner, the admins, members? Protocol-defined, or implementation dependent?
